packages/custom_lint/test/src/v2/server_to_client_channel_test.dart

```diff
@@ -0,0 +1,532 @@
+              '- riverpod v2.1.0\n'
+              '- flutter_hooks v0.18.5\n'
+              '- http_parser v4.0.0\n'
+              '- freezed from path /Users/user/freezed/packages/freezed/\n'
+              '- http_parser from git 4cdfbf9159123746fce29d2862f148f901da66a/\n'
+              '\n'
+              'design_system at /Users/user/project/app/packages/design_system\n'
+              '- riverpod v2.1.1\n'
+              '- flutter_hooks v0.18.5\n'
+              '- freezed from git url ssh://git@github.com/rrousselGit/freezed.git ref 4cdfbf9159f2e9746fce29d2862f148f901da66a path packages/freezed\n'
+              '- http_parser v4.0.0\n'
+              '\n'
+              'This is not supported. Custom_lint shares the analysis between all'
+              ' packages. As such, all plugins are started under a single process,'
+              ' sharing the dependencies of all the packages that use custom_lint. '
+              "Since there's a single process for all plugins, if 2 plugins try to"
+              ' use different versions for a dependency, the process cannot be '
+              'reasonably started. Please make sure all packages have the same version.\n'
+              'You could run the following commands to try fixing this:\n'
+              '\n'
+              'cd /Users/user/project/app\n'
+              'flutter pub upgrade riverpod flutter_hooks freezed\n'
+              'cd /Users/user/project/app/packages/http\n'
+              'flutter pub upgrade riverpod flutter_hooks http_parser freezed http_parser\n'
+              'cd /Users/user/project/app/packages/design_system\n'
+              'flutter pub upgrade riverpod flutter_hooks freezed http_parser',
+            ),
+          ),
+        ),
+      );
+    });
+
+    test('pure dart packages should have simple pub upgrade command', () {
+      final checker = ConflictingPackagesChecker();
+      final firstPubspec = Pubspec('app');
+      final secondPubspec = Pubspec('http');
+      final firstContextRoot = createContextRoot('app');
+      final secondContextRoot = createContextRoot('app/packages/http');
+      final firstContextRootPackages = [
+        createPackage('riverpod', '2.2.0'),
+        createPackage('freezed', '2.3.2'),
+      ];
+      final secondContextRootPackages = [
+        createPackage('riverpod', '2.1.0'),
+        createPackage('freezed', '2.3.1'),
+      ];
+
+      checker.addContextRoot(
+        firstContextRoot,
+        firstContextRootPackages,
+        firstPubspec,
+      );
+      checker.addContextRoot(
+        secondContextRoot,
+        secondContextRootPackages,
+        secondPubspec,
+      );
+
+      expect(
+        checker.throwErrorIfConflictingPackages,
+        throwsA(
+          isA<StateError>().having(
+            (error) => error.message,
+            'message',
+            equals(
+              'Some dependencies with conflicting versions were identified:\n'
```

Comment:
Can you use multiline strings?

Reply:
definitely, I just avoid them because I don't like the way it gets formatted haha